p2p/enode/urlv4.go

```diff
@@ -124,19 +155,17 @@ func parseComplete(rawurl string) (*Node, error) {
 		return nil, fmt.Errorf("invalid node ID (%v)", err)
 	}
 	// Parse the IP address.
-	host, port, err := net.SplitHostPort(u.Host)
+	ips, err := net.LookupIP(u.Hostname())
 	if err != nil {
-		return nil, fmt.Errorf("invalid host: %v", err)
-	}
-	if ip = net.ParseIP(host); ip == nil {
-		return nil, errors.New("invalid IP address")
+		return nil, err
 	}
+	ip = ips[0]
 	// Ensure the IP is 4 bytes long for IPv4 addresses.
 	if ipv4 := ip.To4(); ipv4 != nil {
 		ip = ipv4
 	}
```

Comment:
This logic has changed slight in upstream https://github.com/ethereum/go-ethereum/blob/master/p2p/enode/urlv4.go#L130 we might want to look into syncing it up in this PR. Additionally, a quorum specific comment might be helpful. Something like: 
```
 // Quorum: is only testing the hostname resolves to  a validate the IPv4 addr, but unlike upstream geth, it does not pass the resolved IP on, but rather uses the hostname.
// Parse the IP address.
...
 ```

Reply:
I'm not enthusiastic on pulling in individual upstream changes unless its to fix a critical bug. We will get this change when we upgrade, and we can keep our version (1.8.18) as close to the same version upstream.
As you note in your comment, we don't use the IP anyway, rather passing in the hostname field (which may be an IP).
Let me know if you feel strongly that we do pull this change in.

Reply:
I don't feel strongly, it was only because the change from upstream was being brought in and 
 that had fallen behind (https://github.com/ethereum/go-ethereum/commit/b90cdbaa79cfe438aab0f1389d35980f3d38ec84#diff-0532cc1dc91dc5633d79c93dac1cd87bR128). Removing the logic (as you have now done) seems fine to me and less confusing as it isn't being passed in.